airflow/models.py

```diff
@@ -4001,27 +4025,42 @@ def get_one(
 
         query = (
             session.query(cls.value)
-            .filter(and_(*filters))
-            .order_by(cls.execution_date.desc(), cls.timestamp.desc())
-            .limit(1))
+                .filter(and_(*filters))
+                .order_by(cls.execution_date.desc(), cls.timestamp.desc())
+                .limit(1))
 
         result = query.first()
         if result:
-            return result.value
+            if enable_pickling is None:
+                enable_pickling = configuration.getboolean('core', 'enable_xcom_pickling')
+
+            if enable_pickling:
+                return pickle.loads(result.value)
```

Comment:
We don't really have a way of handling this case, since the migration will need to be manually done by switching the configuration this should be OK (plus the failure mode isn't scary).